### PR TITLE
Updated Visual Basic IDE topic

### LINKED ISSUE #1269: VB Refactoring docs no longer valid for VS2017
Please point to [Refactoring (VB) - https://docs.microsoft.com/en-us/visualstudio/vb-ide/refactoring-vb](https://docs.microsoft.com/en-us/visualstudio/vb-ide/refactoring-vb)

### LINKED ISSUE #1270: VB Code Generation docs should be added for VS2017
VS2017 adds new code generation tooling to VB, docs for which can be found here:  [Code Generation (VB) - https://docs.microsoft.com/en-us/visualstudio/vb-ide/code-generation-vb](https://docs.microsoft.com/en-us/visualstudio/vb-ide/code-generation-vb)

### PR DESCRIPTION
**Pull request by stevehoag**
Added links to Refactoring and Code Generation; removed old Refactoring
link and topic; updated TOC.

Fixes #1269 Fixes #1270

### CODE REVIEW

**Comment by mairaw** [comment]
isn't there a location that you could redirect the file refactoring-and-rename-dialog-box.md instead of simply excluding? the redirection works across repos.

**Comment by stevehoag** [comment]
@mairaw The dialog box no longer exists in VS 2017. Refactoring is done directly in the editor.

File: docs/visual-basic/programming-guide/language-features/declared-elements/declared-element-names.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ Every declared element has a name, also called an *identifier*, which is what th
   
 -   You are working with code written in another language in which the given keyword is not reserved.  
   
- Otherwise, you should consider renaming the element if its name conflicts with a keyword. The integrated development environment (IDE) provides an easy way to do this. For more information, see [Refactoring and Rename Dialog Box](../../../../visual-basic/developing-apps/using-ide/refactoring-and-rename-dialog-box.md).  
+ Otherwise, you should consider renaming the element if its name conflicts with a keyword. The integrated development environment (IDE) provides an easy way to do this. For more information, see [Refactoring](https://docs.microsoft.com/en-us/visualstudio/vb-ide/refactoring-vb.md).
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/language-features/declared-elements/declared-element-names.md]: this link should not have .md in the end or locale in the middle

File: docs/visual-basic/developing-apps/using-ide/using-the-visual-basic-development-environment.md [hunk 2]
```diff
@@ -41,14 +41,19 @@ In addition to the core user interface features that Visual Studio provides, the
...
  [Help for Event Handlers in Visual Basic Code](../../../visual-basic/developing-apps/using-ide/help-for-event-handlers.md)  
  Demonstrates how to get F1 Help on an event handler.  
   
  [How to: Compile and Run a Project](../../../visual-basic/developing-apps/using-ide/how-to-compile-and-run-a-project.md)  
- Describes your options for compiling and running from within the development environment.  
+ Describes your options for compiling and running from within the development environment.
+
+## Related Topics
+
+ [Refactoring](https://docs.microsoft.com/en-us/visualstudio/vb-ide/refactoring-vb)
```
Comment (mairaw) [inline docs/visual-basic/developing-apps/using-ide/using-the-visual-basic-development-environment.md]: same with these links, they shouldn't have locale on them

**Comment by mairaw** [comment]
ok, but 301 redirections are better than 404. shouldn't we redirect folks looking for that topic to https://docs.microsoft.com/visualstudio/vb-ide/refactoring-vb?

File: docs/toc.md [hunk 3]
```diff
@@ -1394,7 +1394,8 @@
 ### [Using the Visual Basic Development Environment](visual-basic/developing-apps/using-ide/using-the-visual-basic-development-environment.md)
 #### [Visual Basic Settings](visual-basic/developing-apps/using-ide/settings.md)
 #### [Visual Basic IntelliSense Code Snippets](visual-basic/developing-apps/using-ide/intellisense-code-snippets.md)
-#### [Refactoring and Rename Dialog Box (visual-basic/developing-apps/Visual Basic)](visual-basic/developing-apps/using-ide/refactoring-and-rename-dialog-box.md)
+#### [Refactoring](https://docs.microsoft.com/en-us/visualstudio/vb-ide/refactoring-vb?toc=dotnet)
```
Comment (mairaw) [inline docs/toc.md]: is this preparing us to have the contextual toc? this doesn't work yet, right? also remove locale from these links on the toc file

File: docs/visual-basic/programming-guide/language-features/declared-elements/declared-element-names.md (hunk 1 above)
Comment (stevehoag) [inline docs/visual-basic/programming-guide/language-features/declared-elements/declared-element-names.md]: Fixed

File: docs/visual-basic/developing-apps/using-ide/using-the-visual-basic-development-environment.md (hunk 2 above)
Comment (stevehoag) [inline docs/visual-basic/developing-apps/using-ide/using-the-visual-basic-development-environment.md]: Fixed

File: docs/toc.md (hunk 3 above)
Comment (stevehoag) [inline docs/toc.md]: Brian had suggested that I try these on Friday, and they didn't work - I forgot that I needed to back out the changes. Thanks!

**Comment by stevehoag** [comment]
@mairaw For the Refactoring Dialog topic, would you suggest leaving the old topic and redirecting to MSDN?

**Comment by mairaw** [comment]
I think they should go to related content in the latest version. If the dialog doesn't exist anymore, shouldn't they just go to that topic you're linking to on the vs docs?

**Comment by stevehoag** [comment]
No, if they're linking to that topic they are most likely using an older version of VS. The instructions in the new Refactoring topics only work on VS 2017.

**Comment by mairaw** [comment]
I'll leave it up to you if you want to redirect people to the old behavior or the new behavior (my vote would be for the new one). The redirect works with full URL redirection too according to the docs.

File: docs/visual-basic/developing-apps/using-ide/using-the-visual-basic-development-environment.md [hunk 4]
```diff
@@ -41,14 +41,19 @@ In addition to the core user interface features that Visual Studio provides, the
...
  [How to: Compile and Run a Project](../../../visual-basic/developing-apps/using-ide/how-to-compile-and-run-a-project.md)  
- Describes your options for compiling and running from within the development environment.  
+ Describes your options for compiling and running from within the development environment.
+
+## Related Topics
+
+ [Refactoring](https://docs.microsoft.com/visualstudio/vb-ide/refactoring-vb)  
+ Describes the the process of improving your code after it has been written by changing the internal structure of the code without changing the external behavior of the code.
+
+ [Code Generation](https://docs.microsoft.com/en-us/visualstudio/vb-ide/code-generation-vb)
```
Comment (mairaw) [inline docs/visual-basic/developing-apps/using-ide/using-the-visual-basic-development-environment.md]: no locale in the URL

**Review by mairaw — APPROVED** [review]
I think this should be good to merge no? Just found a small thing to fix in one of the links

**Pull request merged**